### ISSUE TITLE
How to get VoLTE on MTK 6771 (Umidigi F1)

### ISSUE THREAD

**Issue by sharadmittal** [phhusson/treble_experimentations#405]
I'm able to run latest GSI release on this P60 (MTK 6771) phone. Everything works beautifully except VoLTE. The factory rom supports both VoLTE and VoWifi over T-mobile here in US.

I can see that the factory system image has lots of mediatek ims files. Do we need to copy some of them to our system? Does anyone have any idea on how to fix this?

Thanks.

**Comment by penn5** [comment]
Send me one and I'll try ;). No guarantees. 

Copying the files won't work, they're deeply integrated to the MTK stock ROM.

**Comment by AccountDeletion** [comment]
I'm wondering if anyone has any ideas on this one. I'm starting to think VoLTE on Mediatek is extremely difficult. I've been attempting to copy stock files over, but quite honestly don't really know what I'm doing.

Files I have found and moved from stock rom with no success:

/system/priv-app
ImsService.apk

/system/framework
mediatek-ims-common.jar
mediatek-ims-extension-plugin.jar

/vendor/bin
volte_imcb
volte_imsm_93
volte_stack
volte_ua

Added to build prop
ro.mtk_ims_support=1
ro.mtk_volte_support=1
persist.mtk.volte.enable=1

Again, don't really have any clue what I'm doing, but if anyone has any ideas, I would gladly try them. Thanks in advance

**Comment by phhusson** [comment]
"easy"
Implement IMS Android interface (
https://source.android.com/devices/tech/connect/ims ) with reversed
mediatek HAL
https://github.com/phhusson/vendor_interfaces/blob/pie/vendor/mediatek/hardware/radio/2.0/IRadio.hal

Le jeu. 2 mai 2019 à 20:06, AccountDeletion <notifications@github.com> a
écrit :

**Comment by AccountDeletion** [comment]
Thanks so much for the reply. As I mentioned, in quite over my head, so don't fully comprehend the information or how to implement it... but I'll jump in and get to reading. Hopefully I'll be able to make it work and report back.

Thanks again.

>

**Comment by phhusson** [comment]
Sorry, I forgot the major information
You can find such a project (reimplementing IMS Android framework, using a reverse-engineereed vendor HAL) here:
https://github.com/penn5/hwims-java

I recommend you to come to @phhvolte Telegram group, where you can discuss this with me and @penn5

**Comment by AccountDeletion** [comment]
Great... thanks for the info.

**Comment by sharadmittal** [comment]
We are ready to providing funding for this effort. Coding IMS Service for MTK platforms so that GSI's can do VoLTE and VoWifi. Msg me if anyone is interested.

**Comment by penn5** [comment]
@sharadmittal GitHub doesn't have private messaging

**Comment by penn5** [comment]
Btw vowifi is unlikely to work, ever, although depending on mtk it might be possible

**Comment by penn5** [comment]
I can almost certainly get VoWiFi to work, I just looked at the HALs.

I have started trying to make this but I will definitely need a testing device available, and it will log absolutely everything to start with, including phone numbers.

But its definitely possible

**Comment by sharadmittal** [comment]
> @sharadmittal GitHub doesn't have private messaging

:). I can be reached at @mittals1234 on Telegram

**Comment by Veiranx** [comment]
I'd like to throw my hat into the ring here and mention that I have the same device yet similar situation. My Umidigi F1 on stock firmware/OS works perfectly with Verizon Wireless (USA), yet I don't have any data at all when I use an AOSP-based GSI.

I'm certainly interested in trying the methods mentioned above, and we'll see if they solve my problem, too.

**Comment by sharadmittal** [comment]
No data seems to be an issue with incorrect configuration than anything else. I could be wrong though.

Have you tried playing with your apn settings?

**Comment by Veiranx** [comment]
Yes, though the APN settings look proper (compared to the Pixel 2 that I'm currently using). It's not something I can modify from the Settings menu; everything is greyed out.

Using a root explorer, I tried modifying the system apns_conf.xml and bootlooped the phone (not a problem since I used the SPFT to revert). Attempting to modify the vendor apns_conf.xml didn't really do much; removing a large portion of it lost me even the basic talk+text signal I had. I'm not sure beyond this effect whether certain entries could potentially fix the problem.

**Comment by sharadmittal** [comment]
hmmm. Not sure. You could join this Telegram group and ask if anyone else has the same network/issue.

https://t.me/UMIDIGIF1

**Comment by raj22700** [comment]
Any solution for volte till now on your Phone??
